org.eclipse.m2e.core.ui/src/org/eclipse/m2e/core/ui/archetype/ArchetypeGenerator.java

```diff
@@ -0,0 +1,54 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Konrad Windszus
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *      Konrad Windszus
+ *******************************************************************************/
+package org.eclipse.m2e.core.ui.archetype;
+
+import java.util.Collection;
+import java.util.Properties;
+
+import org.osgi.annotation.versioning.ProviderType;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IProgressMonitor;
+
+import org.eclipse.m2e.core.project.IArchetype;
+import org.eclipse.m2e.core.project.MavenProjectInfo;
+
+
+/**
+ * Implementations of this OSGi service interface allow to create new projects based on Maven archetypes.
+ * 
+ * @since 2.1.0 (package version 1.0.0)
+ */
+@ProviderType
+public interface ArchetypeGenerator {
+
+  /**
+   * Creates project structure using the given archetype and then imports created project(s)
+   *
+   * @return a list of created projects.
+   */
+  default Collection<MavenProjectInfo> createArchetypeProjects(IPath location, IArchetype archetype, String groupId,
+      String artifactId, String version, String javaPackage, Properties properties, IProgressMonitor monitor)
+      throws CoreException
+  {
+    return createArchetypeProjects(location, archetype, groupId, artifactId, version, javaPackage, properties, false,
+        monitor);
+  }
+
+  Collection<MavenProjectInfo> createArchetypeProjects(IPath location, IArchetype archetype, String groupId,
+      String artifactId, String version, String javaPackage, Properties properties, boolean interactive,
```

Comment:
As proposed in #949 for the existing implementation we should consider to use a Map<String, String> instead of Properties. With static Map factories those are much more convenient to handle than Properties if there is a fixed set of key-values.

Reply:
+1, `Map<String, String>` is more modern.

Reply:
fixed by rebasing.